### PR TITLE
feat(datastore): replace to use NewClientWithDatabase on Client creation to…

### PR DESCRIPTION
**Pull request by jwdjj**
… handle both default and non-default database

## Description

For consistency of our [e.g. sample docs](https://cloud.google.com/datastore/docs/concepts/entities)

Main change: Replacing `NewClient(...)` with `NewClientWithDatabase(...)` to handle both default and non-default datastore databases

Other changes:
- Modify test cases accordingly
- Added databaseID variables, no changes on environment variable
- Snippets are still not runnable. Focusing on CRUD entities, no changes on separate queries snippets.
- Add workaround in the MAKEFILE when "gotestsum: command not found" persisted during testing

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [X] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,6 +23,8 @@ test: check-env
 	go install gotest.tools/gotestsum@latest
 	cd ${dir}
 	gotestsum --rerun-fails=3 --packages="./..." --junitfile sponge_log.xml -f standard-verbose -- --timeout 60m
+	# FIX: when the above command failed, run gotestsum without installing
```
Comment (telpirion) [inline Makefile:26]: issue: remove comment.

File: datastore/snippets/snippet_test.go [hunk 2]
```diff
@@ -76,7 +76,7 @@ func SnippetNewKey_withMultipleParents() {
 
 func SnippetClient_Put() {
 	ctx := context.Background()
-	client, _ := datastore.NewClient(ctx, "my-proj")
+	client, _ := datastore.NewClientWithDatabase(ctx, "my-proj", "my-database-id")
```
Comment (telpirion) [inline datastore/snippets/snippet_test.go:79]: issue: use default for tests. Be sure to add a comment that states that the default DB is being used.

Comment (telpirion) [reply datastore/snippets/snippet_test.go:79]: Here and below.

**Review by telpirion — CHANGES_REQUESTED** [review]
Let's make sure that we demonstrate to customers to use the `default` database in our samples. We should assume that default is, well, _the default_ for most use cases.

**Comment by telpirion** [comment]
Hello @jwdjj -- This PR has been open for a couple of weeks without the comments addressed. Do you intend to return to it?

**Comment by jwdjj** [comment]
> Hello @jwdjj -- This PR has been open for a couple of weeks without the comments addressed. Do you intend to return to it?

Acked. yes I'm planning to address my apologies for the delay.

Just one thing: since we don't want to do it on all samples, I feel I need someone from the team to confirm this.

I can revert back and focus on just related to this [snippet](https://cloud.google.com/datastore/docs/concepts/entities#create_an_entity) for consistency on this [document](https://cloud.google.com/appengine/docs/standard/using-firestore-in-datastore-mode?tab=go#using_datastore) since we instruct user to create new datastore and provide the ID. 

Thanks again for reviewing

**Comment by telpirion** [comment]
> Just one thing: since we don't want to do it on all samples, I feel I need someone from the team to confirm this.
> 
> I can revert back and focus on just related to this [snippet](https://cloud.google.com/datastore/docs/concepts/entities#create_an_entity) for consistency on this [document](https://cloud.google.com/appengine/docs/standard/using-firestore-in-datastore-mode?tab=go#using_datastore) since we instruct user to create new datastore and provide the ID.

Please do check with the Datastore/Firestore team. I've included @bhshkh, from the Datastore clients team, on this review.

**Comment by bhshkh** [comment]
> Let's make sure that we demonstrate to customers to use the `default` database in our samples. We should assume that default is, well, _the default_ for most use cases.
> 
> It looks like `NewClient()` is not listed as deprecated ([docs](https://pkg.go.dev/cloud.google.com/go/datastore#NewClient)). That being the case, I think we should change this PR. Instead of refactoring all samples to use `NewClientWithDatabase()`, let's just add 1 new sample that demonstrates how to change the database.

Agree with this
NewClient is not deprecated. Please add new snippet

**Comment by telpirion** [comment]
Thank you for your input, @bhshkh !

Hello @jwdjj -- will you please refactor this PR? Thank you!

**Comment by jwdjj** [comment]
> > Let's make sure that we demonstrate to customers to use the `default` database in our samples. We should assume that default is, well, _the default_ for most use cases.
> > It looks like `NewClient()` is not listed as deprecated ([docs](https://pkg.go.dev/cloud.google.com/go/datastore#NewClient)). That being the case, I think we should change this PR. Instead of refactoring all samples to use `NewClientWithDatabase()`, let's just add 1 new sample that demonstrates how to change the database.
> 
> Agree with this NewClient is not deprecated. Please add new snippet

Acked. Thank you, please let me action this accordingly. Will revert the changes and make the new snippet accordingly.
I will also do the Makefile change as commented above.

**Comment by telpirion** [comment]
Since this PR has been open for over a month and the amount of required work is relatively low, I have simply created the required new sample here: #4529

**Pull request closed without merging**